service/matching/matching_engine.go

```diff
@@ -2038,6 +2037,7 @@ func (e *matchingEngineImpl) updatePhysicalTaskQueueGauge(pqm *physicalTaskQueue
 		float64(loadedPhysicalTaskQueueCounter),
 		metrics.VersionedTag(versioned),
 	)
+	e.gaugeMetrics.lock.Unlock()
```

Comment:
this doesn't make sense. the point of line 2025 is to read from the map under the lock and copy to a local variable, so that we can update the metric outside of the lock. the lock protects the maps only.